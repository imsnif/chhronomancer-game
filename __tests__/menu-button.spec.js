import 'react-native'
import React from 'react'
import MenuButton from '../components/menu-button'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

test('MenuButton (default props) => renders correctly', () => {
  const tree = renderer.create(
    <MenuButton />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('MenuButton (with title) => renders correctly', () => {
  const tree = renderer.create(
    <MenuButton title='foo' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
