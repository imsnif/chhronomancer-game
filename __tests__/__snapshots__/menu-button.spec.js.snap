// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton (default props) => renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 7,
      "width": "100%",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "backgroundColor": "#2f3241",
          "borderColor": "#02ffff",
          "borderRadius": 2,
          "borderWidth": 2,
          "flex": 1,
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#02ffff",
          "paddingTop": 12,
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;

exports[`MenuButton (with title) => renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 7,
      "width": "100%",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "backgroundColor": "#2f3241",
          "borderColor": "#02ffff",
          "borderRadius": 2,
          "borderWidth": 2,
          "flex": 1,
          "width": "100%",
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#02ffff",
          "paddingTop": 12,
          "textAlign": "center",
        }
      }
    >
      foo
    </Text>
  </View>
</View>
`;
